adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -622,6 +624,32 @@ def socket_close(self, socket_num):
         if resp[0][0] != 1:
             raise RuntimeError("Failed to close socket")
 
+    def start_server(self, port, socket_num, conn_mode=TCP_MODE, ip=None): # pylint: disable=invalid-name
+        """Opens a server on the specified port, using the ESP32's internal reference number"""
+        if self._debug:
+            print("*** starting server")
+        self._socknum_ll[0][0] = socket_num
+        port_param = struct.pack('>H', port)
+        if ip:      # use the 4 arg version
+            resp = self._send_command_get_response(_START_SERVER_TCP_CMD,
+                                                   (ip,
+                                                    port_param,
+                                                    self._socknum_ll[0],
+                                                    (conn_mode,)))
+        else:       # use the 3 arg version
+            resp = self._send_command_get_response(_START_SERVER_TCP_CMD,
+                                                   (port_param,
+                                                    self._socknum_ll[0],
+                                                    (conn_mode,)))
+        if resp[0][0] != 1:
+            raise RuntimeError("Could not start server")
+
+    def get_server_state(self, socket_num):
```

Comment:
I think in this situation that wouldn't work, because it takes in a parameter? I will still change it to server_state though, to better match similar status queries (like socket_available, which takes the socknum and returns available bytes)

Reply:
🤦‍♂ Not my most complete thought, obviously. haha